src/wordlift/post-excerpt/class-post-excerpt-rest-controller.php

```diff
@@ -0,0 +1,223 @@
+		$data            = $request->get_params();
+		$post_id         = $data['post_id'];
+		$post_body       = $data['post_body'];
+		$current_hash    = md5( $post_body );
+		$server_response = self::get_post_excerpt_conditionally( $post_id, $post_body, $current_hash );
+		if ( $server_response === NULL || ! array_key_exists( 'post_excerpt', $server_response ) ) {
+			return array(
+				'status' => 'error',
+				'message' => __('Unable to contact wordlift server', 'wordlift')
+			);
+		} else {
+			return array(
+				'status' => 'success',
+				'post_excerpt' => $server_response['post_excerpt'],
+				'from_cache' => $server_response['from_cache'],
+				'message' => __('Excerpt successfully generated.', 'wordlift')
+			);
+		}
+
+	}
+
+	/**
+	 * This call back is invoked by the Rest api action.
+	 */
+	public static function register_route_callback() {
+		/** @var  $post_id_validation_settings array Settings used to validate post id */
+		$post_id_validation_settings   = array(
+			'required'          => TRUE,
+			'validate_callback' => function ( $param, $request, $key ) {
+				return is_numeric( $param );
+			}
+		);
+		$post_body_validation_settings = array(
+			'required'          => TRUE,
+			'validate_callback' => function ( $param, $request, $key ) {
+				return is_string( $param );
+			}
+		);
+		/**
+		 * Rest route for getting the excerpt from wordlift api.
+		 */
+		register_rest_route(
+			WL_REST_ROUTE_DEFAULT_NAMESPACE,
+			'/' . self::POST_EXCERPT_NAMESPACE . '/(?P<post_id>\d+)',
+			array(
+				'methods'             => \WP_REST_Server::CREATABLE,
+				'callback'            => 'Wordlift\Post_Excerpt\Post_Excerpt_Rest_Controller::get_post_excerpt',
+				'permission_callback' => function () {
+					return current_user_can( 'publish_posts' );
+				},
+				'args'                => array(
+					'post_id'   => $post_id_validation_settings,
+					'post_body' => $post_body_validation_settings
+				)
+			)
+		);
+	}
+
+	/**
+	 * This function determines whether to get the excerpt from the server or from the meta cache.
+	 *
+	 * @param $post_id int The Post id.
+	 * @param $post_body string The post content
+	 * @param $current_hash string md5 hash of the current post body.
+	 *
+	 * @return array|bool|null
+	 */
+	public static function get_post_excerpt_conditionally( $post_id, $post_body, $current_hash ) {
+		$previous_data   = get_post_meta( self::POST_EXCERPT_META_KEY, TRUE );
+		$server_response = NULL;
+
+		if ( $previous_data === '' ) {
+			// There is no data in meta, so just fetch the data from remote server.
+			$server_response = self::get_post_excerpt_from_remote_server( $post_id, $post_body );
+		}
+		// If there is data in meta, get the previous hash and compare.
+		$previous_hash = $previous_data['post_body_hash'];
+
+		if ( $current_hash === $previous_hash ) {
+			// then return the previous value.
+			$server_response = array(
+				'post_excerpt' => $previous_data['post_excerpt'],
+				'from_cache' => TRUE
```

Comment:
@naveen17797 

`				'from_cache' => TRUE`

to

`				'from_cache' => true,`